src/homework/sandro/tictactoe/TicTacToeFinal.java

```diff
@@ -91,16 +91,24 @@ public static boolean hasWinInCol(int[][] field, int currentPlayer, int col) {
         return true;
     }
 
-//    public static boolean hasWinInDiagonal(int[][] field, int currentPlayer, int row, int col) {
-//        for (int i = 0; i < field.length; i++) {
-//            for (int j = 0; j < field.length; j++) {
-//                if (i == j) {
-//                    return false;
-//                }
-//            }
-//        }
-//        return true;
-//    }
+    public static boolean hasWinInDiag(int[][] field, int currentPlayer) {
+        for (int iRow = 0; iRow < field.length; iRow++) {
```

Comment:
The name iRow is rather confusing. You are now indexing a column with a variable which has row as part of its name. I would just stick to i here, because neither row nor col makes sense. 

While I really liked the loops you chose to determine whether someone won horizontally or vertically, I think using them here is not the best. Especially hasWinInDiag2 is not very readable. Since your field is limited to 3x3 and you  know that the only possible winning scenarios are field[0, 0] = field[1, 1] = field[2, 2] = currentPlayer OR field[2, 0] = field[1, 1] = field[0, 2] = currentPlayer, I would just create two methods which check exatly this without any fancy loop. Give them a proper name (e.g. hasWinTopLeftToBottomRight, hasWinBottomLeftToTopRight and then hasWonDiagonally() { return hasWinTopLeftToBottomRight || hasWinBottomLeftToTopRight; } <-- in my opinion, that is a lot more readable than fancy loops 😃